src/Hangman/Domain/AggregateRoot/Game.aggregate.ts

```diff
@@ -26,38 +30,99 @@ export class Game extends AggregateRoot {
 
   @Field()
   @IsString()
-  @MinLength(5)
+  @MinLength(3)
   wordToGuess: string;
 
   @Field()
   @IsNumber()
   @Min(1)
   maxGuesses: number;
 
-  constructor({ playerId, wordToGuess, maxGuesses }: GameDto, gameId: string) {
+  @Field((type) => [String])
+  lettersGuessed: string[];
+
+  constructor(id: string, version?: number) {
     super();
-    this.gameId = gameId;
-    this.playerId = playerId;
-    this.wordToGuess = wordToGuess;
-    this.maxGuesses = maxGuesses;
+    this.id = id;
+    this.version = version;
   }
 
-  async startNewGame() {
-    // if we validate here, we can leave validating in CommandConstructor. double validation is unneeded, and here we can async validate using class-validators
+  private logger = new Logger(Game.name);
+
+  async startNewGame(data: GameDto) {
+    // apply to be able to validate
+    this.playerId = data.playerId;
+    this.wordToGuess = data.wordToGuess;
+    this.maxGuesses = data.maxGuesses;
+    this.lettersGuessed = [];
+    this.dateCreated = new Date();
+    this.dateModified = new Date();
 
     try {
       await validateOrReject(this);
 
       this.apply(
         new NewGameStartedEvent(
-          this.gameId,
+          this.id,
           this.playerId,
           this.wordToGuess,
           this.maxGuesses,
+          this.dateCreated,
+          this.dateModified,
         ),
+        false,
       );
     } catch (err) {
       throw new InvalidGameException(err);
     }
   }
+
+  async guessLetter(letter: string) {
+    // TODO: validate guess
+
+    // better validation of course, quick check to see if this works
+    this.lettersGuessed.push(`${letter[0]}`);
+
+    if (this.lettersGuessed?.length - 1 >= this.maxGuesses) {
+      throw new InvalidGameException('Max guesses, game over');
+    }
+
+    const event = new LetterGuessedEvent(this.id, letter, this.lettersGuessed);
```

Comment:
this is not a nice solution, now I have to pass the state of the aggregate in the event (`this.lettersGuessed`), in stead of only the change.
But how to update the view if I don't have the state of the aggregate ?

Use the state of the view, add the new letter and put it back?

Reply:
update: we don't send the state of aggregate in event. in the updater we retrieve the guessed list and add letter to it and save it back